src/reducers/iprove.js

```diff
@@ -44,11 +44,40 @@ const reducer = (state = initialState, action) => {
 
       case NEW_STEP:
         let scope = (key === 'steps') ? newState.currentScope : []
-        depth.splice(index, 0, { scope, dependencies: [], ast: { type: action.payload, ...action.otherArgs } })
+        depth[index] = { scope, dependencies: [], ast: { type: action.payload, ...action.otherArgs } }
         return newState
+<<<<<<< HEAD
+
+      case INSERT_STEP:
+        scope = (key === 'steps') ? newState.currentScope : []
+        depth.splice(index, 0, { scope, dependencies: [], ast: { type: action.payload, ...action.otherArgs } })
+        return { ...newState, steps: newState.steps.filter(Boolean) }
```

Comment:
If we use splice, then we can merge NEW_STEP and INSERT_STEP into one because they are effectively doing the same thing